src/main/java/uk/gov/pay/adminusers/persistence/dao/UserDao.java

```diff
@@ -32,14 +39,47 @@ public UserDao(Provider<EntityManager> entityManager) {
     public List<UserEntity> findByExternalIds(List<String> externalIds) {
         String query = "SELECT u FROM UserEntity u WHERE LOWER(u.externalId) in :externalIds";
 
-        List<String> lowerCaseExternalIds = externalIds.stream().map(String::toLowerCase).collect(Collectors.toList());
+        List<String> lowerCaseExternalIds = externalIds.stream().map(String::toLowerCase).collect(toList());
 
         return entityManager.get()
                 .createQuery(query, UserEntity.class)
                 .setParameter("externalIds", lowerCaseExternalIds)
                 .getResultList();
     }
-
+    
+    public Map<String, List<String>> getAdminUserEmailsForGatewayAccountIds(List<String> gatewayAccountIds) {
+        if (gatewayAccountIds.size() > 0) {
+            String positionalParams = IntStream.rangeClosed(1, gatewayAccountIds.size()).mapToObj(Integer::toString)
+                    .map(i -> "?" + i).collect(Collectors.joining(","));
+
+            String query = "SELECT sga.gateway_account_id, users.email FROM service_gateway_accounts sga" +
+                    " RIGHT JOIN user_services_roles usr" +
+                    " ON usr.service_id = sga.service_id" +
+                    " JOIN users ON users.id = usr.user_id" +
+                    " JOIN roles ON roles.id = usr.role_id" +
+                    " WHERE sga.gateway_account_id in (" + positionalParams + ")" +
+                    " AND roles.name='admin'" +
+                    " ORDER by sga.gateway_account_id";
+
+            Query nativeQuery = entityManager.get().createNativeQuery(query);
+
+            IntStream.rangeClosed(1, gatewayAccountIds.size()).forEach(i ->
+                    nativeQuery.setParameter(i, gatewayAccountIds.get(i - 1)));
+
+            List<Object[]> gatewayAccountIdsToAdminEmails = nativeQuery.getResultList();
+            return gatewayAccountIdsToAdminEmails.stream()
+                    .map(arrayOfObject -> new SimpleEntry<>((String) arrayOfObject[0], (String) arrayOfObject[1]))
```

Comment:
I wonder if this big one liner would be more readable split into a few lines or even a helper method, by the fourth stream I was scratching my head 😄 . I don't think this is a blocker for this functionality